### ISSUE TITLE
2.0.0.beta2 - Custom Layout documentation is wrong

### ISSUE THREAD

**Issue by Phocea** [marmelab/admin-on-rest#1499]
**What you were expecting:**
Trying to use t[he custom layout from the theming documentation](https://github.com/marmelab/admin-on-rest/blob/next/docs/Theming.md) with the react-admin simple example

**What happened instead:**
Getting the error 
> Could not find module in path: 'material-ui/CircularProgress'

If modified with
> import CircularProgress from 'material-ui/Progress';

the rendering still is not working

**Steps to reproduce:**
Run the sandbox:
https://codesandbox.io/s/jjwoj6r4wv

* Admin-on-rest version: 2.0.0.beta2
* React version:
* Browser: chrome
* Stack trace (in case of a JS error):

```
Invariant Violation
Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: undefined. You likely forgot to export your component from the file it's defined in.

Check the render method of `MyLayout`.
```

**Comment by djhi** [comment]
Thanks for reporting :)

**Comment by Phocea** [comment]
Temporary workaround is to "reuse" the react-admin Layout. but hten its not a simplified one anymore ;)

```
// in src/MyLayout.js
import React, { createElement } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import classnames from "classnames";
import {
  MuiThemeProvider,
  createMuiTheme,
  withStyles
} from "material-ui/styles";
import Hidden from "material-ui/Hidden";
import compose from "recompose/compose";

import {
  AppBar,
  Sidebar,
  DRAWER_WIDTH,
  LoadingIndicator,
  Menu,
  Notification,
  defaultTheme
} from "react-admin";

const styles = theme => ({
  root: {
    width: "100%",
    zIndex: 1
  },
  appFrame: {
    position: "relative",
    display: "flex",
    width: "100%",
    height: "100%"
  },
  content: {
    width: "100%",
    marginLeft: 0,
    flexGrow: 1,
    backgroundColor: theme.palette.background.default,
    padding: theme.spacing.unit * 3,
    transition: theme.transitions.create("margin", {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.leavingScreen
    }),
    height: "calc(100% - 56px)",
    [theme.breakpoints.up("md")]: {
      content: {
        height: "calc(100% - 64px)",
        marginTop: 64
      }
    },
    [theme.breakpoints.up("xs")]: {
      marginTop: "4em"
    },
    [theme.breakpoints.down("sm")]: {
      padding: 0
    }
  },
  contentShift: {
    [theme.breakpoints.up("md")]: {
      marginLeft: DRAWER_WIDTH,
      transition: theme.transitions.create("margin", {
        easing: theme.transitions.easing.easeOut,
        duration: theme.transitions.duration.enteringScreen
      })
    }
  }
});

const sanitizeRestProps = ({ staticContext, ...props }) => props;

const MyLayout = ({
  children,
  classes,
  className,
  customRoutes,
  dashboard,
  logout,
  menu,
  open,
  theme,
  title,
  ...props
}) => (
  <MuiThemeProvider theme={createMuiTheme(theme)}>
    <div
      className={classnames("layout", classes.root, className)}
      {...sanitizeRestProps(props)}
    >
      <div className={classes.appFrame}>
        <Hidden xsDown>
          <AppBar title={title} open={open} />
        </Hidden>
        <Sidebar>
          {createElement(menu || Menu, {
            logout,
            hasDashboard: !!dashboard
          })}
        </Sidebar>
        <main
          className={classnames(classes.content, open && classes.contentShift)}
        >
          {children}
        </main>
        <Notification />
        <LoadingIndicator />
      </div>
    </div>
  </MuiThemeProvider>
);

const componentPropType = PropTypes.oneOfType([
  PropTypes.func,
  PropTypes.string
]);

MyLayout.propTypes = {
  children: PropTypes.oneOfType([PropTypes.func, PropTypes.node]),
  classes: PropTypes.object,
  className: PropTypes.string,
  customRoutes: PropTypes.array,
  dashboard: componentPropType,
  logout: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.func,
    PropTypes.string
  ]),
  menu: PropTypes.oneOfType([PropTypes.func, PropTypes.string]),
  open: PropTypes.bool,
  title: PropTypes.node.isRequired,
  theme: PropTypes.object.isRequired
};

MyLayout.defaultProps = {
  theme: defaultTheme
};

const mapStateToProps = state => ({
  open: state.admin.ui.sidebarOpen
});
const enhance = compose(
  connect(
    mapStateToProps,
    {} // Avoid connect passing dispatch in props
  ),
  withStyles(styles)
);

export default enhance(MyLayout);
```

**Comment by djhi** [comment]
`import { CircularProgress } from 'material-ui/Progress';`

https://material-ui-next.com/demos/progress/

**Comment by Phocea** [comment]
Yes I mentionned this fix in the bug report, but then  you get:

```
Invariant Violation
Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: undefined. You likely forgot to export your component from the file it's defined in.

Check the render method of `MyLayout`.
```

**Comment by djhi** [comment]
Nope, read again and note the `{ CircularProgress }`. I  just tried it on last beta, it workd

**Comment by djhi** [comment]
Here you go: https://codesandbox.io/s/p93jnopvpm

**Comment by Phocea** [comment]
Ahh yes, the infamous {} 👍

**Comment by fzaninotto** [comment]
Fixed by #1501
